Bundler/Viewer/Unity/Assets/Scripts/Network/Scripts/NetworkAnchor.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using Persistence;
+
+public class NetworkAnchor : MonoBehaviour
+{
+    /// <summary>
+    /// The game object to enable when the initial network anchor is being loaded.
+    /// </summary>
+    public GameObject LoadingAnchorRoot;
+
+    /// <summary>
+    /// The game object to enable when the initial network anchor has been loaded.
+    /// </summary>
+    public GameObject FoundAnchorRoot;
+
+    /// <summary>
+    /// For this to function, the game object must also have the AnchorPersistence behavior applied.
+    /// </summary>
+    private AnchorPersistence anchorPersistence;
+
+    /// <summary>
+    /// For this to function, there must be a global NetworkAnchorManager.
+    /// </summary>
+    private NetworkAnchorManager networkAnchorManager;
+
+    /// <summary>
+    /// For this to function, the game object must also have a local instance of an NetworkAnchorPlayer.
+    /// </summary>
+    private NetworkAnchorPlayer anchorPlayer;
+
+    private PersistenceEventArgs pendingPersistenceEventArgs = null;
+
+    private void Awake()
+    {
+        if (FoundAnchorRoot != null)
```

Comment:
is it ever ok for *AnchorRoot to be null?

Reply:
Yes, these aren't needed for the shared anchors to function.   These are only needed if you wants to show/hide objects during the initial loading of the anchors